lib/core/viewer/components/viewer.component.ts

```diff
@@ -694,7 +693,22 @@ export class ViewerComponent implements OnChanges, OnInit, OnDestroy {
     }
 
     onSubmitFile(newImageBlob: Blob) {
-        this.fileSubmit.emit(newImageBlob);
+        const newImageFile: File = new File([newImageBlob], this?.nodeEntry?.entry?.name, { type: this?.nodeEntry?.entry?.content?.mimeType });
```

Comment:
you can move this line inside the if-block as there's a chance it is not needed based on the condition below